lib/rspec/mocks/any_instance/expect_chain_chain.rb

```diff
@@ -24,6 +24,7 @@ def create_message_expectation_on(instance)
 
         def invocation_order
           @invocation_order ||= {
+            :with => [nil],
             :and_return => [nil],
             :and_raise => [nil],
             :and_yield => [nil]
```

Comment:
You make an excellent point, maybe we should change how this is accessed instead...